compact.cpp

```diff
@@ -32,7 +32,7 @@ unsigned __stdcall ObjectMemory::lastOTEntry()
 	const OTE* pOT = m_pOT;
 	while (pOT[i].isFree())
 	{
-		ASSERT(i >= 0);
+		ASSERT(0 < i);
```

Comment:
I'm happy to change the direction or remove the assertion. It seems that leaving the assert is reasonable as evidence that we believe the integer will not wrap. It only exists in the Debug configuration, so isn't a performance issue otherwise.